packages/mobile/src/screens/edit-track-screen/fields/ReleaseDateField.tsx

```diff
@@ -1,135 +1,24 @@
-import { useCallback, useMemo, useRef, useState } from 'react'
-
 import type { Nullable } from '@audius/common'
-import { Theme } from '@audius/common'
 import { useField } from 'formik'
 import moment from 'moment'
-import { TouchableOpacity } from 'react-native-gesture-handler'
-import type {
-  CustomCancelButtonPropTypes,
-  CustomConfirmButtonPropTypes
-} from 'react-native-modal-datetime-picker'
-import DateTimePickerModal from 'react-native-modal-datetime-picker'
-
-import { Button, Pill, Text } from 'app/components/core'
-import { makeStyles } from 'app/styles'
-import { useThemeColors, useThemeVariant } from 'app/utils/theme'
 
-const isToday = (date: Date) => moment(date).isSame(moment(), 'day')
+import { ContextualMenu } from 'app/components/core'
 
 const messages = {
   label: 'Release Date',
   today: 'Today'
 }
 
-const useStyles = makeStyles(({ spacing }) => ({
-  root: {
-    flexDirection: 'row',
-    justifyContent: 'space-between',
-    alignItems: 'center',
-    paddingHorizontal: spacing(6),
-    paddingVertical: spacing(2)
-  },
-  dateText: {
-    textTransform: 'uppercase'
-  },
-  datePickerModal: {
-    // Specific padding to hide the underlying "done" button with the "cancel" button
-    paddingBottom: 37
-  }
-}))
+export const ReleaseDateField = (props) => {
+  const [{ value }] = useField<Nullable<string>>('release_date')
 
-const ConfirmDateButton = (props: CustomConfirmButtonPropTypes) => {
-  const { label, onPress } = props
   return (
-    <Button
-      style={{ borderRadius: 0 }}
-      variant='primary'
-      size='large'
-      title={label}
-      fullWidth
-      onPress={onPress}
-      pressScale={1}
+    <ContextualMenu
+      menuScreenName='ReleaseDate'
+      label={messages.label}
+      value={value}
+      pillValue={value ? moment(value).format('M/D/YY @ h:mm A') : null}
```

Comment:
Is this not what `renderValue` is for?

Reply:
i wanted to use the default renderValue without directly using the form's value since it needed formatting. but maybe overriding renderValue is the cleaner way.

<img width="341" alt="Screenshot 2023-12-15 at 8 29 38 AM" src="https://github.com/AudiusProject/audius-protocol/assets/6413636/b9a680d7-029f-4667-a798-a8b39d6a2fe9">

Reply:
actually i think just renaming pillValue -> formattedValue would make this clearer and seems re-usable.